src/vm/zapsig.cpp

```diff
@@ -1324,19 +1324,44 @@ BOOL ZapSig::EncodeMethod(
     {
         Module * pReferencingModule = pMethod->IsNDirect() ?
             pMethod->GetModule() :
-            (Module *)pResolvedToken->tokenScope;
+            IsDynamicScope(pResolvedToken->tokenScope) ? NULL: GetModule(pResolvedToken->tokenScope);
 
-        if (!pReferencingModule->IsInCurrentVersionBubble())
+        if (pReferencingModule != NULL && !pReferencingModule->IsInCurrentVersionBubble())
         {
             // FUTURE: Encoding of new cross-module references for ReadyToRun
             // This warning is hit for recursive cross-module inlining. It is commented out to avoid noise.
             // GetSvcLogger()->Printf(W("ReadyToRun: Method reference outside of current version bubble cannot be encoded\n"));
             ThrowHR(E_FAIL);
         }
 
-        methodToken = pMethod->IsNDirect() ?
-            pMethod->GetMemberDef_NoLogging() :
-            pResolvedToken->token;
+        if (pMethod->IsNDirect())
+        {
+            methodToken = pMethod->GetMemberDef_NoLogging();
+        }
+        else if (!IsDynamicScope(pResolvedToken->tokenScope))
+        {
+            // Normal case for IL code
+            methodToken = pResolvedToken->token;
+        }
+        else
+        {
+            // Dynamic scope case. IL Stubs only
```

Comment:
Any way we can assert this instead of commenting on it? (i.e. "IL Stubs only" <- can that be asserted?)